src/canvas/renderCanvas.js

```diff
@@ -488,9 +568,9 @@ export const renderCanvas = memoize(
 
     renderBackgroundFills(context, canvasWidth, canvasHeight);
 
-    renderReactMarksAndMeasures(
+    const schedulerAreaEndY = renderReactMarksAndMeasures(
```

Comment:
It's really quite unintuitive that `renderReactMarksAndMeasures` returns its minY. I understand why that needs to be computed there, but it should be more visible, as I only realised this is what was happening while reviewing `2/n`

`renderReactMarksAndMeasuresAndReturnLaneHeight`? 😜

Reply:
Update from `3/n`, now that both `renderReactEventRow` and `renderReactMeasures` return a new Y, the "stacking" logic is much clearer than it was when it was just one function. 

Wouldn't got as far as renaming them anymore, but I'd definitely annotate their function signatures with `: number` to make it obvious from the start they return something, and add a comment explaining what they return.

Reply:
Yeah, my idea was for all these render functions to report what they rendered so that we can lay everything out based on what was rendered.

I think this isn't super clear because we're rendering in overlapping layers, and not all layers have stacks. We're also stacking exclusively in the Y direction, but there's definitely a possibility of stacking in the X direction as well (e.g. for the old priority labels).

I'll add the `: number` in this PR.

There's also the unrelated issue of having to recalculate all of these in the `getHoveredEvent` util function. Ideally `getHoveredEvent` will be able to reuse the calculations from these render functions so that we can be more confident that the returned hovered events are actually the ones rendered on screen.